catalog/spatial/csw/spatial-csw-source-common/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/common/source/AbstractCswSource.java

```diff
@@ -955,26 +958,31 @@ public ResourceResponse retrieveResource(
       CswRecordCollection recordCollection;
       try {
         recordCollection = csw.getRecordById(getRecordByIdRequest, rangeValue);
+      } catch (CswException e) {
+        throw new ResourceNotFoundException(
+            String.format(ERROR_ID_PRODUCT_RETRIEVAL, metacardId), e);
+      }
 
-        Resource resource = recordCollection.getResource();
-        if (resource != null) {
+      Resource resource = recordCollection.getResource();
+      if (resource != null) {
 
-          long responseBytesSkipped = 0L;
-          if (recordCollection.getResourceProperties().get(BYTES_SKIPPED) != null) {
-            responseBytesSkipped =
-                (Long) recordCollection.getResourceProperties().get(BYTES_SKIPPED);
-          }
-          alignStream(resource.getInputStream(), requestedBytesToSkip, responseBytesSkipped);
+        long responseBytesSkipped = 0L;
+        if (recordCollection.getResourceProperties().get(BYTES_SKIPPED) != null) {
+          responseBytesSkipped = (Long) recordCollection.getResourceProperties().get(BYTES_SKIPPED);
+        }
+
+        try (InputStream resourceIs = resource.getInputStream()) {
```

Comment:
After looking at this closer, I don't think we can put this stream in the try-with-resource. When the try finishes without error it will still close the stream but the caller of this method is expecting to use that stream to read the resource. We could still close the stream if there is an exception but shouldn't close it otherwise.